### PR TITLE
[release/6.0] Make sure WinHttpHandler 6.0.1 is released

### LINKED ISSUE #60291: Subsequent Grpc calls with WinHttpHandler to an IIS hosted service fail
### Description

I'm trying to call a Grpc service hosted in IIS over http2 from NetFramework app. After reading the relevant docs([supported platforms](https://docs.microsoft.com/en-us/aspnet/core/grpc/supported-platforms?view=aspnetcore-5.0#net-grpc-client-requirements)) and issues (https://github.com/dotnet/aspnetcore/issues/9020, https://github.com/dotnet/core/issues/5713) I understand that this should be possible if I'm using Win11/WinServer2022 with the pre release version of the System.Net.Http.WinHttpHandler nuget package (in my case 6.0.0-rc.1.21451.13).

Unfortunately this seems not to be the case, as only the first call to the service succeeds and any subsequent call fails with `Status(StatusCode="Cancelled", Detail="No grpc-status found on response. Using gRPC with WinHttp has Windows and package version requirements. See https://aka.ms/aspnet/grpc/netstandard for details.")`.

Things get even stranger when I tested combining calls made with the NetCore's SocketsHttpHandler. A call with the WinHttpHandler will fail no matter whether the first call to the service was made with WinHttp or Sockets, and it appears that the problem is somehow triggered by the state of the IIS application as restarting the host resets the issue(restarting the client has no effect). Furthermore even after the WinHttp call has failed one can successfully call the service with the Sockets handler from the same process.

Here is a table with the different setups
| Client        | Server        | Result |
| ------------- | ------------- | ------ |
|WinHttp | IIS | :x: Fail |
|WinHttp | Kestrel | :heavy_check_mark: Succeeds |
|Sockets | IIS | :heavy_check_mark: Succeeds |
|Sockets | Kestrel | :heavy_check_mark: Succeeds |

### Reproduction Steps

I've created a minimal project that reproduces the problem - https://github.com/angelyordanov/grpc-iis-winhttp-sample

1. Start the GrpcService with the IISExpress profile
2. Start either one of the console apps

### Expected behavior

There should be no exceptions and all grpc calls should succeed.

### Actual behavior

The second call made with the WinHttpHandler always fails.

### Regression?

No

### Known Workarounds

No

### Configuration

```
dotnet --info
.NET SDK (reflecting any global.json):
 Version:   5.0.401
 Commit:    4bef5f3dbf

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.401\
```

### Other information

_No response_

### PR DESCRIPTION
**Pull request by antonfirsov**
This addresses an overlook from #63346. We forgot to follow [the nuget servicing guideline](https://github.com/dotnet/runtime/blob/main/docs/project/library-servicing.md#how-to-service-a-library) so the bugfix was not released.

Fixes #60291

/cc @safern

## Customer Impact

This is follow up to PR #63346 which was approved and merged for 6.0.2, but missed nuget authoring -- as a result, the fix didn't really ship as part of 6.0.2.

#60291 is a bug in WinHttpHandler that prevents running/debugging gRPC code on .NET Framework using Visual Studio + IIS Express which is an important customer scenario for https://github.com/dotnet/core/issues/5713. It can probably also manifest in production environments with servers other than Kestrel.

## Testing

N/A

## Risk

Low.

### CODE REVIEW

**Review by safern — APPROVED** [review]

**Comment by antonfirsov** [comment]
@danmoseley does this need to go through servicing process?

**Comment by danmoseley** [comment]
No approval needed. This is part of the original approved fix

**Review by karelz — APPROVED** [review]

**Review by wfurt — APPROVED** [review]
LGTM

**Comment by karelz** [comment]
@carlossanlop it does not need Tactics approval -- see above. Not sure if you want to put servicing-approved on it, or if it is ok this way.

**Comment by carlossanlop** [comment]
Added the label, per Dan's comment that it does not need approval.
I noticed one CI failure in wasm related to System.Text.Json, so I assume it's unrelated. I'll run it once more just to be 100% safe.
Looks good to merge.

**Question**: @ericstj @bartonjs This kind of change is _only_ made in a servicing branch, correct? It's not expected to see such change in `main`? I ask because this PR was not created as a backport, and I wonder if nuget packages for `main` get published similarly.

**Review by bartonjs — APPROVED** [review]

**Comment by bartonjs** [comment]
> This kind of change is only made in a servicing branch, correct? It's not expected to see such change in main? I ask because this PR was not created as a backport, and I wonder if nuget packages for main get published similarly or differently.

In `main` all of the libraries that are distributed on NuGet already have `<IsPackable>true</IsPackable>` (https://github.com/dotnet/runtime/search?q=IsPackable).

Somewhere in the process of setting up servicing branches, all of the `IsPackable`s are set to false (or deleted, or something).  During servicing it needs to be manually asserted.

`ServicingVersion` is always 0 for `main` builds, so it's never present in csproj files in main.  It's just the count of times that a package has been serviced during that major version.  (The package version is BranchMajor.BranchMinor.ServicingVersion, so for here, 6.0.1.  For main, it's the more complicated BranchMajor.BranchMinor.(ServicingVersion=0)-previewN(-maybe more stuff?).

**Comment by ericstj** [comment]
> Somewhere in the process of setting up servicing branches, all of the IsPackables are set to false (or deleted, or something). During servicing it needs to be manually asserted.

IIRC It's statically defined to give different behavior when PreReleaseVersionLabel is servicing.  Once that's set we default to not build the package, unless individual projects opt-in.

**Comment by safern** [comment]
Yup, this is where we do it:

https://github.com/dotnet/runtime/blob/a7a4ed959365851d0f9f38f5dfb9b2fe7ef04a93/eng/packaging.targets#L21-L29

**Pull request merged**
